bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -545,34 +534,37 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
       send(repMsg.get(), clientId);
     }
   }
-  delete m;
+  delete msg;
 }  // namespace bftEngine::impl
 
 template <>
-void ReplicaImp::onMessage<preprocessor::PreProcessResultMsg>(preprocessor::PreProcessResultMsg *m) {
+void ReplicaImp::onMessage<preprocessor::PreProcessResultMsg>(std::unique_ptr<preprocessor::PreProcessResultMsg> msg) {
   LOG_DEBUG(GL,
             "Handling PreProcessResultMsg via ClientRequestMsg handler "
-                << KVLOG(m->clientProxyId(), m->getCid(), m->requestSeqNum()));
-  return onMessage<ClientRequestMsg>(m);
+                << KVLOG(msg->clientProxyId(), msg->getCid(), msg->requestSeqNum()));
+  return onMessage<ClientRequestMsg>(std::make_unique<ClientRequestMsg>(msg.release()));
 }
 
 template <>
-void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(ReplicaAsksToLeaveViewMsg *m) {
+void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(std::unique_ptr<ReplicaAsksToLeaveViewMsg> message) {
+  auto *msg = message.release();
```

Comment:
i'd delete this line

Reply:
For now, there is no further handling of unique message pointers in the code. That's why all of them get released in all onMessage functions. The addition of proper handling is one of the next stages.